qctrlopencontrols/dynamic_decoupling_sequences/driven_controls.py

```diff
@@ -270,6 +273,20 @@ def convert_dds_to_driven_control(
                                   extras={'deduced_pulse_start_timing': pulse_start_ends[:, 0],
                                           'deduced_pulse_end_timing': pulse_start_ends[:, 1]})
 
+    # check if the minimum_segment_duration is respected in the gaps between the pulses
+    if not np.any(np.logical_or(np.greater(pulse_start_ends[1:, 0]-pulse_start_ends[:-1, 1],
+                                           minimum_segment_duration),
+                                np.isclose(pulse_start_ends[1:, 0], pulse_start_ends[:-1, 1]))):
```

Comment:
Shouldn't this be checking `np.isclose(pulse_start_ends[1:, 0]-pulse_start_ends[:-1, 1], minimum_segment_duration)`?

May be a little more readable if before you start this `if` check you make a local var for the gap durations:
`gap_durations = pulse_start_ends[1:, 0]-pulse_start_ends[:-1, 1]`